django/template/base.py

```diff
@@ -74,23 +74,25 @@
 VARIABLE_ATTRIBUTE_SEPARATOR = '.'
 BLOCK_TAG_START = '{%'
 BLOCK_TAG_END = '%}'
+_BLOCK_TAG_PATTERN = rf'{BLOCK_TAG_START}.*?{BLOCK_TAG_END}'
 VARIABLE_TAG_START = '{{'
 VARIABLE_TAG_END = '}}'
+_VARIABLE_TAG_PATTERN = rf'{VARIABLE_TAG_START}.*?{VARIABLE_TAG_END}'
 COMMENT_TAG_START = '{#'
 COMMENT_TAG_END = '#}'
+_COMMENT_TAG_PATTERN = rf'{COMMENT_TAG_START}.*?{COMMENT_TAG_END}'
 SINGLE_BRACE_START = '{'
 SINGLE_BRACE_END = '}'
 
 # what to report as the origin for templates that come from non-loader sources
 # (e.g. strings)
 UNKNOWN_SOURCE = '<unknown source>'
 
-# match a variable or block tag and capture the entire tag, including start/end
-# delimiters
-tag_re = (_lazy_re_compile('(%s.*?%s|%s.*?%s|%s.*?%s)' %
-          (re.escape(BLOCK_TAG_START), re.escape(BLOCK_TAG_END),
-           re.escape(VARIABLE_TAG_START), re.escape(VARIABLE_TAG_END),
-           re.escape(COMMENT_TAG_START), re.escape(COMMENT_TAG_END))))
+# match a block, variable, or comment tag and capture the entire tag, including
+# start/end delimiters re.compile is faster than instantiating _lazy_re_compile
+tag_re = re.compile(rf'({_BLOCK_TAG_PATTERN}'
+                    rf'|{_VARIABLE_TAG_PATTERN}'
+                    rf'|{_COMMENT_TAG_PATTERN})')
```

Comment:
I would go back to a single string and use a comment like I was suggesting. This just overcomplicates things unnecessarily.